python/tank/util/yaml_cache.py

```diff
@@ -18,17 +18,124 @@
 import threading
 
 from tank_vendor import yaml
-from ..errors import TankError
+from ..errors import (
+    TankError,
+    TankUnreadableFileError,
+    TankFileDoesNotExistError,
+)
+
+class CacheItem(object):
+    """
+    Represents a single item in the global yaml cache.
+
+    Each item carries with it a set of data, an stat from the .yml file that
+    it was sourced from (in os.stat form), and the path to the .yml file that
+    was sourced.
+    """
+
+    def __init__(self, path, data=None, stat=None):
+        """
+        Initializes the item.
+
+        :param path:    The path to the .yml file on disk.
+        :param data:    The data sourced from the .yml file.
+        :param stat:    The stat of the file on disk. If not provided, an os.stat
+                        will be run and the result stored.
+        :raises:        tank.errors.TankUnreadableFileError: File stat failure.
+        """
+        self._path = path
+        self._data = data
+        try:
+            self._stat = stat or os.stat(path)
+        except Exception, exc:
+            raise TankUnreadableFileError("Unable to stat file '%s': %s" % (path, exc))
+
+    @property
+    def data(self):
+        """The item's data."""
+        return self._data
+
+    @data.setter
```

Comment:
The answer is a resounding "no" ;)

I tweaked it to work properly in 2.5 in my local repo. Next push will contain the fix.